lib/mix/lib/mix/utils.ex

```diff
@@ -338,7 +340,13 @@ defmodule Mix.Utils do
         when node: term()
   def write_dot_graph!(path, title, nodes, callback, _opts \\ []) do
     {dot, _} = build_dot_graph(make_ref(), nodes, MapSet.new(), callback)
-    File.write!(path, ["digraph ", quoted(title), " {\n", dot, "}\n"])
+    contents = ["digraph ", quoted(title), " {\n", dot, "}\n"]
+
+    if path == "-" do
+      IO.write(contents)
+    else
+      File.write!(path, contents)
+    end
```

Comment:
This can sort of be a breaking change for other tasks that use `write_dot_graph!/5`, but I don't think it matters much and we can probably add `--output` to those tasks too.